chain/indexer/src/streamer/mod.rs

```diff
@@ -116,15 +118,44 @@ async fn build_streamer_message(
 
         let shard_id = header.shard_id.clone() as usize;
 
-        indexer_shards[shard_id].receipt_execution_outcomes = receipt_outcomes
-            .into_iter()
-            .map(|IndexerExecutionOutcomeWithOptionalReceipt { execution_outcome, receipt }| {
-                IndexerExecutionOutcomeWithReceipt {
-                    execution_outcome,
-                    receipt: receipt.expect("`receipt` must be present at this moment"),
+        let mut receipt_execution_outcomes: Vec<IndexerExecutionOutcomeWithReceipt> = vec![];
+        for outcome in receipt_outcomes {
+            let IndexerExecutionOutcomeWithOptionalReceipt { execution_outcome, receipt } = outcome;
+            let receipt = if let Some(receipt) = receipt {
+                receipt
+            } else {
+                // Receipt might be missing only in case of delayed local receipt
+                // that appeared in some of the previous blocks
+                // we will be iterating over previous blocks until we found the receipt
+                'find_local_receipt: loop {
+                    let prev_block =
+                        match fetch_block_by_hash(&client, block.header.prev_hash).await {
```

Comment:
@khorolets It seems that this loop will always query only one block before the one that the streamer works with. It should be `prev_block.header.prev_hash` instead of `block.header.prev_hash`. This will lead to an endless loop if there is no receipt in the previous block.

Reply:
oops